planning/behavior_velocity_planner/src/utilization/path_utilization.cpp

```diff
@@ -169,20 +177,24 @@ autoware_auto_planning_msgs::msg::Path interpolatePath(
   }
 
   // Calculate query points
-  // Remove query point if query point is near input path point
-  std::vector<double> s_tmp = s_in;
+  // Use all values of s_in to inherit the velocity-planned point, and add some points for
+  // interpolation with a constant interval if the point is not closed to the original s_in points.
+  // (because if this interval is very short, the interpolation will be less accurate and may fail.)
+  std::vector<double> s_out = s_in;
+
+  const auto has_almost_same_value = [&](const auto & vec, const auto x) {
+    if (vec.empty()) return false;
+    const auto has_close = [&](const auto v) { return std::abs(v - x) < epsilon; };
+    return std::find_if(vec.begin(), vec.end(), has_close) != vec.end();
+  };
   for (double s = 0.0; s < path_len; s += interval) {
-    s_tmp.push_back(s);
-  }
-  std::sort(s_tmp.begin(), s_tmp.end());
-
-  for (const double s : s_tmp) {
-    if (!s_out.empty() && std::fabs(s_out.back() - s) < epsilon) {
-      continue;
+    if (!has_almost_same_value(s_out, s)) {
+      s_out.push_back(s);
     }
-    s_out.push_back(s);
   }
 
+  std::sort(s_out.begin(), s_out.end());
```

Comment:
@TakaHoribe 
is this sort meaningful?

Reply:
since the s_out is not sorted in order, without this sort, the interpolation fails.